netsim/tools/nuts/tests/test_bgp.j2

```diff
@@ -0,0 +1,43 @@
+{% set ns = namespace(found=false) %}
+- test_class: TestNapalmBgpNeighbors
+  test_data:
+{% for nn,nd in nodes.items() %}
+{%   for neighbor in nd.bgp.neighbors %}
+{%     if "ipv4" in neighbor %}
+{%     set ns.found = true %}
```

Comment:
If we assume that every neighbor has at least one AF (and I think that's a pretty safe assumption, otherwise I'll prune the neighbor list in the BGP module), then you could change the top loop to `for nn,nd in nodes.items() if nd.bgp.neighbors` and then use the `else` clause of the FOR loop to generate `[]`. No pressure though, this is OK.

Reply:
Yeah, I know why you're using ns.found. I'm just saying that if **nd.bgp.neighbors** has some usable value you'll get some usable data in the test (which is why I said you could use the for/else loop).

Never mind, it's perfectly OK the way it is